osquery/distributed/plugins/bro_distributed.cpp

```diff
@@ -0,0 +1,336 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+// clang-format off
+// This must be here to prevent a WinSock.h exists error
+#include "osquery/remote/transports/tls.h"
+// clang-format on
+
+#include <vector>
+#include <sstream>
+
+#include <boost/property_tree/ptree.hpp>
+
+#include <osquery/config.h>
+#include <osquery/database.h>
+#include <osquery/distributed.h>
+#include <osquery/enroll.h>
+#include <osquery/flags.h>
+#include <osquery/registry.h>
+#include <osquery/distributed.h>
+
+#include "osquery/core/json.h"
+#include "osquery/remote/serializers/json.h"
+#include "osquery/remote/utility.h"
+
+#include "osquery/remote/bro/BrokerManager.h"
+#include "osquery/remote/bro/QueryManager.h"
+#include "osquery/remote/bro/utils.h"
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+
+FLAG(string, bro_ip, "localhost", "IP address of bro (default localhost)")
+
+FLAG(uint64, bro_port, 9999, "Port of bro (default 9999)")
+
+FLAG(string, bro_groups, "{}", "List of groups (default {})")
+
+class BRODistributedPlugin : public DistributedPlugin {
+ public:
+  Status setUp() override;
+
+  Status getQueries(std::string& json) override;
+
+  Status writeResults(const std::string& json) override;
+
+ private:
+  std::vector<std::string> startup_groups_;
+};
+
+REGISTER(BRODistributedPlugin, "distributed", "bro");
+
+Status BRODistributedPlugin::setUp() {
+  // Setup Broker Endpoint
+  LOG(INFO) << "Starting the Bro Distributed Plugin";
+  broker::init();
+  BrokerManager& bm = BrokerManager::get();
+
+  // Subscribe to all and individual topic
+  Status s_mq_all = bm.createMessageQueue(bm.TOPIC_ALL);
```

Comment:
If you're not using these `Status` objects simultaneously you can reuse/copy-assign the same local variable.

```cpp
auto s =  bm.createMessageQueue(bm.TOPIC_ALL);
if (!s.ok()) {
  ...
}

s = bm.createMessageQueue(bm.TOPIC_PRE_INDIVIDUALS + bm.getNodeID());
....
```